datafusion/expr-common/src/statistics.rs

```diff
@@ -857,6 +857,143 @@ pub fn compute_variance(
     ScalarValue::try_from(target_type)
 }
 
+/// Merges two distributions into a single distribution that represents their combined statistics.
+/// This creates a more general distribution that approximates the mixture of the input distributions.
```

Comment:
I think it would help to explain in comments what assumptions can be made from the combined distribution

For example, is it guaranteed that the `range` is conservative (as in it is known that there are no values that lay outside the range?)

Though now that I ask it seems like maybe we need to clarify if the range of `GenericDistribution` is conservative 🤔 

https://github.com/apache/datafusion/blob/8a2e83eb74f89a4e3387817943749f3894e7141a/datafusion/expr-common/src/statistics.rs#L273-L272

Reply:
Maybe we can add some notes to `GenericDistribution`

```
    /// # Range Guarantees
    /// The provided range is assumed to be conservative - all possible values of the
    /// distribution must lie within this range. However, the range itself might be
    /// an approximation, as the actual distribution could occupy only a subset of the range.
```

Reply:
I added some notes for the method: https://github.com/apache/datafusion/pull/15296/commits/5b98f4c5efe5404a6ae4c8609a3574cccc91e9ab